lib/src/Uncertainty/Algorithm/WeightedExperiments/SmolyakExperiment.cxx

```diff
@@ -0,0 +1,348 @@
+  LOGDEBUG(OSS() << "  combinationIndicesCollectionSize = " << combinationIndicesCollectionSize);
+  // Fill the indices
+  IndicesCollection combinationIndicesCollection(combinationIndicesCollectionSize, dimension);
+  UnsignedInteger multiindexIndex = 0;
+  const UnsignedInteger strataIndexMin = (level_ < dimension) ? 0 : level_ - dimension;
+  LOGDEBUG(OSS() << "  strataIndexMin = " << strataIndexMin);
+  LOGDEBUG(OSS() << "  level_ = " << level_);
+  for (UnsignedInteger strataIndex = strataIndexMin; strataIndex < level_; ++ strataIndex)
+  {
+    LOGDEBUG(OSS() << "  strataIndex = " <<  strataIndex);
+    const UnsignedInteger strataCardinal = enumerateFunction.getStrataCardinal(strataIndex);
+    const UnsignedInteger cumulatedCardinal = enumerateFunction.getStrataCumulatedCardinal(
+          strataIndex
+        );
+    const UnsignedInteger indexStart = cumulatedCardinal - strataCardinal;
+    for (UnsignedInteger i = indexStart; i < cumulatedCardinal; ++ i)
+    {
+      Indices indices(enumerateFunction(i));
+      for (UnsignedInteger j = 0; j < dimension; ++ j) indices[j] += 1;
+      LOGDEBUG(OSS() << "  indices = " <<  indices);
+      std::copy(indices.begin(), indices.end(), combinationIndicesCollection.begin_at(multiindexIndex));
+      ++ multiindexIndex;
+    } // loop over the indices in the strata
+  } // loop over the strata
+  return combinationIndicesCollection;
+}
+
+/* Comparison class with std:map interface. */
+class PointApproximateComparison
+{
+public:
+  PointApproximateComparison(const Scalar absoluteEpsilon,
+                             const Scalar relativeEpsilon)
+    : absoluteEpsilon_(absoluteEpsilon)
+    , relativeEpsilon_(relativeEpsilon)
+  {
+    // Nothing to do
+  };
+
+  /* Compare two points, according to lexicographic order
+  * Returns true if x < y, false otherwise.
+  */
+  bool operator()(const Point x, const Point y) const
+  {
+    const UnsignedInteger dimension = x.getDimension();
+    if (y.getDimension() != dimension) throw InvalidArgumentException(HERE) << "Error: the two points must have the same dimension. Here x has dimension " << dimension << " while y has dimension " << y.getDimension();
+    bool comparison = false;
+    for (UnsignedInteger k = 0; k < dimension; ++ k)
+    {
+      // std::cout << "  " << x[k] << " < " << y[k] << " ?" << std::endl;
+      const Scalar maximumXY = std::max(std::abs(x[k]), std::abs(y[k]));
+      const Scalar delta = absoluteEpsilon_ + relativeEpsilon_ * maximumXY;
+      if (x[k] + delta < y[k])
+      {
+        comparison = true;
+        break;
+      }
+      else if (x[k] > y[k] + delta)
+      {
+        break;
+      }
+    } // Loop over the dimensions
+    return comparison;
+  }
+private:
+  // Absolute tolerance for comparison
+  Scalar absoluteEpsilon_;
+  // Relative tolerance for comparison
+  Scalar relativeEpsilon_;
+};
+
+// Implement merge with std::map
+void SmolyakExperiment::mergeNodesAndWeights(
```

Comment:
first argument on the same line as the method name, second argument on the second line